interfaces/dbus/backend.go

```diff
@@ -57,14 +57,23 @@ func (b *Backend) Name() interfaces.SecuritySystem {
 // setupDbusServiceForUserd will setup the service file for the new
 // `snap userd` instance on re-exec
 func setupDbusServiceForUserd(snapInfo *snap.Info) error {
-	coreRoot := snapInfo.MountDir()
+	coreOrSnapdRoot := snapInfo.MountDir()
+
+	// fugly - but we need to make sure that the content of the "snapd"
+	// snap wins
+	if snapInfo.InstanceName() == "core" && osutil.FileExists(filepath.Join(coreOrSnapdRoot, "../..", "snapd/current")) {
```

Comment:
Isn't this potentially racing against doLinkSnap() ?

Reply:
mmh, yes, that's  problematic, and relates to the issue that link-snap and setup-profiles are separate but should be merged, that's a long standing pending issue

Reply:
So atm the files in core and snap are identical as @mvo5 mentioned,  so this can probably land after adding to this something like:

     // TODO:  this is racy but the content of the files in core and snapd is identical
     //  cleanup after link-snap and setup-profiles are unified

for extra points we could add a spread test that would fail  if the files in snapd diverge?